### PR TITLE
Enforce function arguments when parsing functional instructions.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by axic** [comment]
So far my understanding was that every instruction can be written in functional notation. This changes that into only instructions which take no arguments can be written in non-functional notation.

I think this may be a breaking change.

**Comment by axic** [comment]
If we are doing a breaking change, I'd rather just prefer eliminating:
- non-functional notation altogether
- literal pushes to the stack

**Comment by chriseth** [comment]
I hope this is not a breaking change, at least on my machine, the tests did not break. Note that this change only applies to the `parseExpression` function which is only invoked if we already are in a functional context like in `let x := <...>` or `mul(<...>, 2)`.

File: Changelog.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ Features:
 
 Bugfixes:
  * Unused variable warnings no longer issued for variables used inside inline assembly.
+ * Inline Assembly: Enforce function arguments when parsing functional instructions.
```
Comment (axic) [inline Changelog.md]: Would add a note, perhaps in the documentation, that functional instructions are enforced within higher level constructs (vardecl, function call, etc.)

**Review by axic — CHANGES_REQUESTED** [review]
Would add a note, perhaps in the documentation, that functional instructions are enforced within higher level constructs (vardecl, function call, etc.)

**Comment by chriseth** [comment]
Clarified.

**Review by axic — APPROVED** [review]

**Pull request merged**